engine/src/operators/OrderBy.cpp

```diff
@@ -221,6 +205,52 @@ std::vector<cudf::table_view> partition_table(const ral::frame::BlazingTableView
 	return cudf::experimental::split(sortedTable.view(), split_indexes);
 }
 
+std::unique_ptr<ral::frame::BlazingTable> generate_partition_plan(const std::vector<ral::frame::BlazingTableView> & samples, 
+	std::size_t table_num_rows, std::size_t avg_bytes_per_row, const std::string & query_part, Context * context){
+	std::vector<cudf::order> sortOrderTypes;
+	std::vector<int> sortColIndices;
+	cudf::size_type limitRows;
+	std::tie(sortColIndices, sortOrderTypes, limitRows) = get_sort_vars(query_part);
+
+	std::unique_ptr<ral::frame::BlazingTable> partitionPlan;
+	
+	std::size_t num_bytes_per_order_by_partition = 400000000; 
```

Comment:
this pattern feels vless than optimal It basically means if you have a config that is used in multiple places you have N number of places for it to mess up and it requires a kind of dirty check if this exists first. A much cleaner way of handling this would be setting the defaults in the config_options map in context. That way you only hnave to do it once and you never have to check for it to exist. The other upside to this is that its backwards compatible with all the code you have written thus far. You can just stop checkingt for its existence and 
```
do std::size_t num_bytes_per_order_by_partition = context->getConfigOptions()["CONFIG_NAME"] 
```

you know this will work because you will have set the default at the beginning. An even better implementation would take the map you brought in from python and convert it to a struct so you can do something like context->getNumBytesOrderBy or context->config()->getNumBytes...

Reply:
its 11 lines of code that become 2

Reply:
This could be considered out of scope for this pr but im used to just looking at everything i see in green and analyzing it